VMEncryption/main/patch/UbuntuPatching.py

```diff
@@ -58,8 +58,10 @@ def __init__(self, logger, distro_info):
         self.touch_path = '/usr/bin/touch'
 
     def install_adal(self):
-        self.command_executor.Execute('apt-get update')
-        self.command_executor.Execute('apt-get install -y python-pip')
+        return_code = self.command_executor.Execute('apt-get install -y python-pip')
+        if return_code != 0:
+            self.command_executor.Execute('apt-get -o Acquire::ForceIPv4=true -y update')
```

Comment:
add a comment explaining what is being done here, and a log message indicating that this path was taken